airbyte-config/init/src/main/java/io/airbyte/config/init/RepositorySeed.java

```diff
@@ -0,0 +1,119 @@
+ */
+
+package io.airbyte.config.init;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.base.Charsets;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.UUID;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+/**
+ * This class takes in a yaml file with a list of objects. It then then assigns each object a uuid
+ * based on its name attribute. The uuid is written as a field in the object with the key specified
+ * as the id-name. It then writes each object to its own file in the specified output directory.
+ * Each file's name is the generated uuid. The goal is that a user should be able to add objects to
+ * the database seed without having to generate uuids themselves. The output files should be
+ * compatible with our file system database (config persistence). It also checks that each name is
+ * unique in the set it is outputting to avoid duplicates clobbering each other.
+ */
+public class RepositorySeed {
+
+  private static final Options OPTIONS = new Options();
+  private static final Option ID_NAME_OPTION = new Option("id", "id-name", true, "field name of the id");
+  private static final Option INPUT_PATH_OPTION = new Option("i", "input-path", true, "path to input file");
+  private static final Option OUTPUT_PATH_OPTION = new Option("o", "output-path", true, "path to where files will be output");
+  static {
+    ID_NAME_OPTION.setRequired(true);
+    INPUT_PATH_OPTION.setRequired(true);
+    OUTPUT_PATH_OPTION.setRequired(true);
+    OPTIONS.addOption(ID_NAME_OPTION);
+    OPTIONS.addOption(INPUT_PATH_OPTION);
+    OPTIONS.addOption(OUTPUT_PATH_OPTION);
+  }
+
+  public void run(String idName, Path input, Path output) throws IOException {
+    final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    final JsonNode jsonNode = mapper.readTree(input.toFile());
+    final Iterator<JsonNode> elements = jsonNode.elements();
+    final Set<String> names = new HashSet<>();
+
+    while (elements.hasNext()) {
+      final JsonNode element = Jsons.clone(elements.next());
+      final String name = element.get("name").asText();
+
+      // validate the name is unique.
+      if (names.contains(name)) {
+        throw new IllegalArgumentException("Multiple records have the name: " + name);
+      }
+      names.add(name);
+
+      final UUID uuid = UUID.nameUUIDFromBytes(name.getBytes(Charsets.UTF_8));
```

Comment:
i'm on the fence on this. i like the idea of being able to generate the uuids deterministically from the configs in some way. the issue is that if someone ever changes the name of a connector then the uuid will change. right now i don't think that causes any immediate problems, but it could be really confusing in our version migration story. if a connector name ever changes we need to make sure its uuid gets propagated to any already persisted config that uses the old one. a few options... not in love with any of these:
1. name field is never allowed to change. if a display name needs to change we add that as an optional attribute. (bad because not immediately intuitive to a contributor)
1. generate uuid based off of position in the list. if an integration is ever removed then a placeholder must be left in the list. (bad because not immediately intuitive to a contributor, and cluttering up the list with placeholders seems not great)
1. keep forcing users to hardcode uuids and just put them in the yaml. validate that all uuids are unique (still generating uuids is annoying, but ultimately this is the most bullet proof solution, so if we can't think of something better, it might be the right one).

Reply:
what about we use the image name for the uuid? (`airbyte/destination-bigquery`) ?

It gives us two things:
- if someone tweaks the integrations and publishes it on their image repo, it automatically gets a new uuid
- it is deterministic

Reply:
I'm okay with this. It is the same problem as using name though. It means we better never change the underlying image name that we are using! I think the cases where that might be something we do is in cases like the file source, were we have considered exposing it at multiple separate sources but all with the same image. there are things we could do to tweak the name, but it seems like weird incidental complexity to add.

Reply:
If that so then we should have the uuid in the yaml. That gives us the best of both world for the small cost of generating a random uuid.

Reply:
the name as more chances to change because it is just a matter of modifying an input whereas the docker repo is a bit more involve. so whichever we choose we should pick the one that is the most stable.
But in the end nothing beats forcing the uuid in the yaml